symfit/core/minimizers.py

```diff
@@ -31,7 +32,18 @@ def __init__(self, objective, parameters):
         """
         self.parameters = parameters
         self._fixed_params = [p for p in parameters if p.fixed]
-        self.objective = partial(objective, **{p.name: p.value for p in self._fixed_params})
+        if isinstance(objective, BaseObjective):
+            self.objective = objective
+        else:
+            # Minimize the provided custom objective instead. This why want to
+            # minimize a CallableNumericalModel, thats what they are for.
+            from .fit import CallableNumericalModel
```

Comment:
Import higher up

Reply:
Unfortunately it has to be here to prevent cyclic imports. Perhaps we can consider moving models to a seperate module in a seperate PR and then this import can be performed normally.

Reply:
Makes sense. I am in favour of splitting some files, but that is indeed for later